schema_utils.py

```diff
@@ -387,3 +411,89 @@ def is_valid_email(obj):
             bool. Whether the given object is a valid email.
         """
         return bool(re.search(r'^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$', obj))
+
+    @staticmethod
+    def is_valid_expression(obj, algebraic=True):
+        """Checks if the given  obj (a string) represents a valid algebraic or
+        numeric expression.
+
+        Args:
+            obj: str. The given expression.
+            algebraic: bool. True if the given expression is algebraic
+            else numeric.
+
+        Returns:
+            bool. Whether the given object is a valid expression.
+        """
+        valid_expression = True
+        valid_characters = '({[]})+-/*^.'
+
+        # Expression should not contain invalid characters.
+        for character in obj:
+            if character not in valid_characters and not bool(
+                    re.match(r'[a-zA-Z0-9]', character)):
+                valid_expression = False
+                break
+
+        # Algebraic expressions should contain at least one latin letter and
+        # numeric expressions should contain none.
+        if algebraic:
+            valid_expression &= bool(re.search(r'[a-zA-Z]+', obj))
+        else:
+            valid_expression &= not bool(re.search(r'[a-zA-Z]+', obj))
+
+        # Expression should contain a balanced bracket sequence.
+        valid_expression &= contains_balanced_brackets(obj)
+
+        # Expression should not contain multiple operators consecutively.
+        valid_operators = valid_characters[6:]
+        valid_expression &= not any(
+            obj[i] in valid_operators and obj[i - 1] in valid_operators for i in
+            python_utils.RANGE(1, len(obj)))
+
+        # Expression should not be an equation or inequality.
+        valid_expression &= not bool(re.search(r'(=|<|>)+', obj))
+
+        return valid_expression
+
+    @staticmethod
+    def is_each_element_a_single_latin_letter(obj):
+        """Returns True iff all elements of the given object (a list) are
+        singular latin letters (uppercase or lowercase).
+
+        Args:
+            obj: list(*). A list of strings.
+
+        Returns:
+            bool. Whether the given object has all latin letters.
+        """
+        for ele in obj:
+            if not bool(re.search(r'[a-zA-Z]+', ele)) or len(ele) != 1:
+                return False
+        return True
+
+    @staticmethod
+    def is_valid_math_equation(obj):
+        """Checks if the given  obj (a string) represents a valid math equation.
+
+        Args:
+            obj: str. A string.
+
+        Returns:
+            bool. Whether the given object is a valid math equation.
+        """
+        if obj.count('=') != 1:
+            return False
+
+        is_valid_expression = get_validator('is_valid_expression')
+        lhs, rhs = obj.split('=')
+
+        # Both sides have to be valid expressions and at least one of them has
+        # to be a valid algebraic expression.
```

Comment:
2 = 6 - 4 seems like a valid equation to me...

Reply:
That's true, but I was under the impression that we won't be supporting purely numeric equations. Also, would purely numeric expressions ever be useful?

Reply:
Yes. E.g. questions about decomposition into place values: 734 = 700 + 30 + 4.

Reply:
If I'm not wrong, this would fall under the numeric expression interaction, won't it? From the creator's POV, I think the question would be formed like, "Expand 734 into its component place values" and the accepted answer would be 700 + 30 + 4 which would use the numeric expressions interaction. 
Also, if we are willing to support purely numeric equations as well, then wouldn't we have to split the math equation interaction into algebraic and numeric equation interactions (the same reason for splitting the expressions interaction would apply here as well)?
But that again brings up the question of how useful would the numeric equation interaction be?

Reply:
Fair points. Let's leave this for now, but I'll ask the question I had above, which is: if we need to expand this later to include numeric equations too, what steps would need to be taken?

Reply:
The same points mentioned above apply here as well. We could simply allow purely numeric equations but the rule-set is going to be built for algebraic equations only, so it wouldn't work as expected. To support that we would probably have to create a new interaction for numeric equations, though again, not sure how useful that would be.